tests/TestCase/ORM/Association/BelongsToManySaveAssociatedOnlyEntitiesAppendTest.php

```
@@ -17,69 +17,48 @@
 namespace Cake\Test\TestCase\ORM\Association;
 
 use Cake\Datasource\ConnectionManager;
+use Cake\Datasource\EntityInterface;
 use Cake\ORM\Association\BelongsToMany;
 use Cake\ORM\Entity;
 use Cake\ORM\Table;
 use Cake\TestSuite\TestCase;
+use Exception;
 
 /**
  * Tests BelongsToManySaveAssociatedOnlyEntitiesAppendTest class
  */
 class BelongsToManySaveAssociatedOnlyEntitiesAppendTest extends TestCase
 {
     /**
-     * @var \Cake\ORM\Table|\PHPUnit\Framework\MockObject\MockObject
+     * Test that saveAssociated() ignores non entity values.
      */
-    protected $tag;
+    public function testSaveAssociatedOnlyEntitiesAppend(): void
+    {
+        $connection = ConnectionManager::get('test');
+        $table = new class (['alias' => 'Tags', 'table' => 'tags', 'connection' => $connection]) extends Table {
+            public function saveAssociated()
+            {
+                throw new Exception('Should not be called');
```

I wonder if using exceptions as pseudo assertions is the right thing to do.

---

It should generally work as a replacement for `never()`. It could end up silently passing when methods higher in the stack are swallowing errors. Another approach would be to set properties on the test case that can have assertions on the values.

---

In general I am afraid of some failure case not being caught because of removal of the assertions we get with mocks.

Just because an exception is not thrown doesn't always mean something worked as expected.